### DISCUSSION TITLE
High availability (HA, multiple bot instances)

### DISCUSSION THREAD

**Discussion by slavafomin** [telegraf/telegraf#1537] [Q&A]
Hello!

I'm trying to scale my application horizontally, each application instance starts a Telegraf bot in a webhook mode. All ingress requests are handled by a single reverse-proxy and distributed among the available instances. However, I'm experiencing a situation where Telegraf fails to work, requests are just not getting processed. I'm also seeing the following error in a log: `terminated by setWebhook request`.

Is it possible to scale Telegraf horizontally? Thanks.

**Comment by NormandoHall** [comment]
You can't distributed among all the instances, because only one can handle each request. Maybe you can do round-robin fashion in nginx proxy, to balance al the requests, but handle one at a time. If you use session, also you will need to sticky with other shared resources between instances, like redis, etc.

**Comment by TrurlMcByte** [comment]
In my case, it's working, but with unlimited repeating last update, balanced between servers. 
It looks like Telegram doesn't get confirmation. 
But with one server same code works fine...

**Reply by MKRhere** [reply]
Could you describe your setup? I'd like to recreate it locally in a similar fashion to test.

**Reply by TrurlMcByte** [reply]
it's integrated with express. Now I have a nearly working solution with the next code:
```js
      const app = express();
      const telegraf = new Telegraf(process.env.BOT_TOKEN);
      const someFixedPath= `/telegraf/my-some-fixed-path-for-all-servers`;

...
      app.post(someFixedPath, async (req, res) => {
        try {
          await telegraf.handleUpdate(req.body);
        } finally {
          res.status(204).end();
        }
      });
...
      telegraf.telegram
          .setWebhook(process.env.WEBHOOK_URL + someFixedPath)
          .then(() => console.log("Webhook is set on", process.env.WEBHOOK_URL))
          .catch(error => console.error("telegraf setWebhook", error));
```
But sometimes getting telegram error: ```429: too many requests: retry after 1``` on one of the servers without any updates or other activity.
